linkerd/proxy/http/src/client.rs

```diff
@@ -67,141 +65,129 @@ pub enum ClientServiceFuture {
 
 // === impl Layer ===
 
-pub fn layer<T, B>(h2_settings: crate::h2::Settings) -> Layer<T, B>
+pub fn layer<B>(h2_settings: crate::h2::Settings) -> Layer<B>
 where
     B: hyper::body::Payload + Send + 'static,
 {
     Layer {
         h2_settings,
-        _p: PhantomData,
+        _marker: PhantomData,
     }
 }
 
-impl<T, B> Clone for Layer<T, B>
+impl<B> Clone for Layer<B>
 where
     B: hyper::body::Payload + Send + 'static,
 {
     fn clone(&self) -> Self {
         Self {
             h2_settings: self.h2_settings,
-            _p: PhantomData,
+            _marker: self._marker,
         }
     }
 }
 
-impl<T, C, B> tower::layer::Layer<C> for Layer<T, B>
+impl<C, B> tower::layer::Layer<C> for Layer<B>
 where
-    Client<C, T, B>: tower::Service<T>,
     B: hyper::body::Payload + Send + 'static,
 {
-    type Service = Client<C, T, B>;
+    type Service = MakeClient<C, B>;
 
     fn layer(&self, connect: C) -> Self::Service {
-        Client {
+        MakeClient {
             connect,
             h2_settings: self.h2_settings,
-            _p: PhantomData,
+            _marker: PhantomData,
         }
     }
 }
 
 // === impl Client ===
 
-/// MakeService
-impl<C, T, B> tower::Service<T> for Client<C, T, B>
+impl<C, T, B> tower::Service<T> for MakeClient<C, B>
 where
     C: tower::MakeConnection<T> + Clone + Send + Sync + 'static,
     C::Future: Send + 'static,
     <C::Future as Future>::Error: Into<Error>,
     C::Connection: Send + 'static,
-    T: connect::HasPeerAddr + HasSettings + fmt::Debug + Clone + Send + Sync,
+    T: HasSettings + fmt::Debug + Clone + Send + Sync,
     B: hyper::body::Payload + 'static,
 {
-    type Response = ClientService<C, T, B>;
+    type Response = Client<C, T, B>;
     type Error = Error;
-    type Future = ClientNewServiceFuture<C, T, B>;
+    type Future = MakeFuture<C, T, B>;
 
     fn poll_ready(&mut self) -> Poll<(), Self::Error> {
         Ok(().into())
     }
 
-    fn call(&mut self, config: T) -> Self::Future {
-        debug!("building client={:?}", config);
-        let peer_addr = config.peer_addr();
-
+    fn call(&mut self, target: T) -> Self::Future {
+        trace!("Building HTTP client");
```

Comment:
I notice we no longer include the target in this event — is that because it's already present in the span context?

Reply:
yeah. there's loads of redundant info now that we have spans on the stack.